pkg/store/consul/consulutil/watch.go

```diff
@@ -327,7 +328,7 @@ func WatchDiff(
 				mapCopy[key] = val
 			}
 
-			outgoingChanges := &WatchedChanges{}
+			outgoingChanges := &WatchedChanges{Total: len(pairs)}
```

Comment:
Maybe the thing to do is, instead, to make `WatchDiff` not emit an output (and not update its state) if the total is 0. This will affect all users of `WatchDiff` though. Which coincidentally is only `dsstore` today.